pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1246,6 +1260,28 @@ func (aws *AWSCloud) DeleteVolume(volumeName string) error {
 	return awsDisk.deleteVolume()
 }
 
+// Implements Volumes.GetVolumeLabels
+func (c *AWSCloud) GetVolumeLabels(volumeName string) (map[string]string, error) {
+	awsDisk, err := newAWSDisk(c, volumeName)
+	if err != nil {
+		return nil, err
+	}
+	info, err := awsDisk.getInfo()
+	if err != nil {
+		return nil, err
+	}
+	labels := make(map[string]string)
+	az := aws.StringValue(info.AvailabilityZone)
+	if az != "" {
```

Comment:
See comments above.  I'd prefer these empty strings to be replaced by explicit errors or nil.